app/src/main/java/com/alif/newsreader/activities/MainActivity.java

```diff
@@ -250,36 +234,51 @@ public boolean onOptionsItemSelected(MenuItem item) {
                 .setPriority(Priority.LOW)
                 .build()
                 .getAsString(new StringRequestListener() {
                     @Override
                     public void onResponse(String response) {
+
                         try {
-                            newsFeedList.clear();
-                            JSONObject responseObj = XML.toJSONObject(response);
-                            JSONObject rssObj = responseObj.getJSONObject("rss");
-                            JSONObject channelObj = rssObj.getJSONObject("channel");
-                            JSONArray detailsArr = channelObj.getJSONArray("item");
-                            for (int i = 0; i < detailsArr.length(); i++) {
-                                JSONObject childJSONObject = detailsArr.getJSONObject(i);
-                                GoogleFeed googleFeed = new GoogleFeed();
-                                googleFeed.setNewsTitle(childJSONObject.getString("title"));
-                                googleFeed.setLink(childJSONObject.getString("link"));
-                                googleFeed.setNewsCategory(childJSONObject.getString("category"));
-                                googleFeed.setDescription(childJSONObject.getString("description"));
-                                newsFeedList.add(googleFeed);
-                                Log.d(DEBUG_TAG, googleFeed.toString());
-                            }
+                            InputStream stream = new ByteArrayInputStream(response.getBytes("UTF-8"));
+
+                            // get list of news feeds
+                            newsFeedList = googleNewsXmlParser.parse(stream);
+
+                            adapter = new NewsFeedAdapter(MainActivity.this, newsFeedList);
+
+                            //This is the code to provide a sectioned list
+                            List<SimpleSectionedRecyclerViewAdapter.Section> sections = new ArrayList<>();
+
+                            // News Feed Header
+                            if (newsFeedList.size() != 0)
+                                sections.add(new SimpleSectionedRecyclerViewAdapter.Section(0, newsFeedList.get(0).getNewsCategory()));
+
+                            //Add your adapter to the sectionAdapter
+                            SimpleSectionedRecyclerViewAdapter.Section[] sectionArr = new SimpleSectionedRecyclerViewAdapter.Section[sections.size()];
+                            SimpleSectionedRecyclerViewAdapter mSectionedAdapter = new
+                                    SimpleSectionedRecyclerViewAdapter(MainActivity.this, R.layout.section, R.id.section_text, adapter);
+                            mSectionedAdapter.setSections(sections.toArray(sectionArr));
+
+                            //Apply mSectionedAdapter adapter to the RecyclerView
+                            newsFeedRecyclerView.setAdapter(mSectionedAdapter);
+
+                            Log.d(DEBUG_TAG, "" + newsFeedList.size());
```

Comment:
Log.d(DEBUG_TAG, "MainActivity::fetchGoogleNewsFeed newsFeedList size is:" + newsFeedList.size());